config/v1/types_infrastructure.go

```diff
@@ -1082,9 +1082,39 @@ type VSpherePlatformStatus struct {
 	LoadBalancer *VSpherePlatformLoadBalancer `json:"loadBalancer,omitempty"`
 }
 
+// IBMCloudServiceEndpoint stores the configuration of a custom url to
+// override existing defaults of IBM Cloud Services.
+type IBMCloudServiceEndpoint struct {
+	// Name is the name of the IBM Cloud service.
+	// examples of these services are:
+	//   IAM=https://cloud.ibm.com/apidocs/iam-identity-token-api
+	//   ResourceController=https://cloud.ibm.com/apidocs/resource-controller/resource-controller
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
```

Comment:
I'll check on that and add some more doc. The expectation is that a service endpoint can only be added once (one `IAM` service endpoint), but could include none to many services. Perhaps there is a way to add validation/tests to enforce that.

Reply:
I don't have a max length for the name in mind, but it would likely be pretty short, perhaps a max of 25 or even 16 chars is sufficient. While I expect there is a finite list of IBM Cloud Services that will require and allow specific overridden endpoints, since I have not yet completed that investigation and development, I cannot provide that finite list at this time. Knowing API changes take a long time to pull in, and using this abstract ServiceEndpoint model from other platforms, should allow flexibility to complete.

Reply:
We have a list of the endpoints in the parent right? You can make that a list map
```
// +listType=map
// +listMapKey=name
```
This will prevent duplicate names being specified in the list

Reply:
That should be added now.